lib/THC/THCCachingAllocator.cpp

```diff
@@ -158,12 +158,12 @@ struct THCCachingAllocator
     allocated_blocks.erase(it);
 
```

Comment:
i think @colesbury had some reservations about this as well. this can be greatly simplified. i'll let him comment on monday.